File: Matlab/Oct_15.m
% Comparing alpha and log(1/(1-alpha))
a = (0.01:0.005:0.1);
y = log(1./(1-a));
plot(a); hold on; plot(y); hold off;

% Computing p-values for N independent hypothesis tests
N = 100;
n = 10;
alpha = 0.05;
p_values = zeros(N, 1);
for i = 1:N
    % Testing the following:
    % H0: true_lambda >= lambda0 vs. H1: true_lambda < lambda0 
    true_lambda = 1.7;
    lambda0 = 2;
    X = poissrnd(true_lambda, 1, n);
    T = (mean(X) - lambda0)/(sqrt(mean(X)/n));
    p_value = normcdf(T);
    p_values(i) = p_value;
end

% Single Test
plot(1:N, p_values, '.')
hold on;
plot(1:N, alpha*ones(N, 1))
hold off;
title('Single Tests')
xlabel('Sample')
ylabel('P-value')

% Bonferroni Test
plot(1:N, p_values, '.')
hold on;
plot(1:N, alpha/N*ones(N, 1))
hold off;
title('Bonferroni Tests')
xlabel('Sample')
ylabel('P-value')

% BH Test
[sorted_p_values, ind]= sort(p_values);
plot(1:N, sorted_p_values, '.')
hold on;
plot(1:N, alpha/N*(1:N))
hold off;
title('BH Tests')
xlabel('Sample')
ylabel('P-value')

File: R/Oct_15.R
# Comparing alpha and log(1/(1-alpha))
a <- seq(.005, .1, .001)
y <- log(1/(1-a))
plot(a, type="l", col='red', ylab="alpha")
lines(y, type="l", col='blue')

# Computing p-values for N independent hypothesis tests
N <- 100
n <- 10
alpha <- .05
p_values <- rep(0, N)
for (i in 1:N){
  # Testing the following
  # H0: lambda >= 2 vs. H1: lambda < 2
  true_lambda <- 1.5
  lambda0 <- 2
  X <- rpois(n, true_lambda)
  T <- (mean(X) - lambda0)/(sqrt(mean(X)/n))
  p_value <- pnorm(T)
  p_values[i] <- p_value
}

# Single Test
plot(1:N, p_values, cex=.4, ylab="p-value",  xlab="Sample", title("Single Test"))
lines(1:N, rep(alpha, N), col='red')

# Bonferroni Test
plot(1:N, p_values, cex=.4, ylab="p-value", xlab="Sample", title("Single Test"))
lines(1:N, rep(alpha/N, N), col='red')

# BH Test
sorted_p_values = sort(p_values)
plot(1:N, sorted_p_values, cex=.4, ylab="p-value", xlab="Sample", title("Single Test"))
lines(1:N, alpha*(1:N)/N, col='red')


